test/package/test_directory_reader.py

```diff
@@ -0,0 +1,180 @@
+
+            pe.save_text("two", "f.txt", "hello, f!")
+            pe.save_text("two", "g.txt", "hello, g!")
+
+        import zipfile
+
+        zip_file = zipfile.ZipFile(filename, "r")
+
+        with TemporaryDirectory() as temp_dir:
+            zip_file.extractall(path=temp_dir)
+            importer = PackageImporter(str(Path(temp_dir) / Path(filename).name))
+            reader_one = importer.get_resource_reader("one")
+            with self.assertRaises(FileNotFoundError):
+                reader_one.resource_path("a.txt")
+
+            self.assertTrue(reader_one.is_resource("a.txt"))
+            self.assertEqual(
+                reader_one.open_resource("a.txt").getbuffer(), b"hello, a!"
+            )
+            self.assertFalse(reader_one.is_resource("three"))
+            reader_one_contents = list(reader_one.contents())
+            self.assertSequenceEqual(
+                reader_one_contents, ["a.txt", "b.txt", "c.txt", "three"]
+            )
+
+            reader_two = importer.get_resource_reader("two")
+            self.assertTrue(reader_two.is_resource("f.txt"))
+            self.assertEqual(
+                reader_two.open_resource("f.txt").getbuffer(), b"hello, f!"
+            )
+            reader_two_contents = list(reader_two.contents())
+            self.assertSequenceEqual(reader_two_contents, ["f.txt", "g.txt"])
+
+            reader_one_three = importer.get_resource_reader("one.three")
+            self.assertTrue(reader_one_three.is_resource("d.txt"))
+            self.assertEqual(
+                reader_one_three.open_resource("d.txt").getbuffer(), b"hello, d!"
+            )
+            reader_one_three_contenst = list(reader_one_three.contents())
+            self.assertSequenceEqual(reader_one_three_contenst, ["d.txt", "e.txt"])
+
+            self.assertIsNone(importer.get_resource_reader("nonexistent_package"))
+
+    def test_scriptobject_failure_message(self):
+        """
+        Test basic saving and loading of a ScriptModule in a directory.
+        Currently not supported.
+        """
+        from package_a.test_module import ModWithTensor
+
+        scripted_mod = torch.jit.script(ModWithTensor(torch.rand(1, 2, 3)))
+
+        filename = self.temp()
+        with PackageExporter(filename, verbose=False) as e:
+            e.save_pickle("res", "mod.pkl", scripted_mod)
+
+        import zipfile
+
+        zip_file = zipfile.ZipFile(filename, "r")
+
+        with self.assertRaisesRegex(
+            RuntimeError,
+            "Loading ScriptObjects from a PackageImporter created from a "
+            "directory is not supported. Use a package archive file instead.",
+        ):
+            with TemporaryDirectory() as temp_dir:
+                zip_file.extractall(path=temp_dir)
+                dir_importer = PackageImporter(
+                    str(Path(temp_dir) / Path(filename).name)
+                )
+                dir_mod = dir_importer.load_pickle("res", "mod.pkl")
+
```

Comment:
I think it would be useful to test that the `importlib.resources` API works with `DirectoryReader` as well. See the rest of the tests in `test/package/test_resources.py`.